spring-cloud-aws-sns/src/main/java/io/awspring/cloud/sns/sms/core/SnsSmsOperations.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sns.sms.core;
+
+import io.awspring.cloud.sns.sms.attributes.SmsMessageAttributes;
+
+/**
+ * @author Matej Nedic
+ * @since 3.0.0
+ */
+public interface SnsSmsOperations {
```

Comment:
Perhaps we can think about changing this interface somehow to contains just two methods and the method parameters would make a distinction if it goes to phone number, topic or target, but I don't have now solution in mind.

Reply:
We could make a special class that would have TopicArn, TargetArn or a Phone Number and just accept that.